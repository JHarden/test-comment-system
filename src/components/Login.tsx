import React from "react";
import styled from "styled-components";

export interface LoginProps {
    userName: string | undefined;
    onLogin: (value: string) => void;
}

const StyledLogin = styled.div`
    background: #dedcdc;
    height: 60px;
    display: flex;
    justify-content: flex-end;
    padding-right: 20px;
    margin: 20px;
`;

const StyledLoginInput = styled.input`
    font-size: 14px;
    min-width: 80px;
    height: 30px;
    border-radius: 5px;
    border: none;
    padding: 0 10px;
`
const LoginButton = styled.button`
    font-size: 20px;
    padding: 5px 0;
    background: #4ca4ff;
    position: relative;
    left: 20px;
    border-radius: 5px;
    color: #fff;
    width: 60px;
    height: 35px;
    cursor: pointer;
    position: relative;
    top: 3px;
`;

const WelcomeMessage = styled.div`
    color: grey;
    font-size: 22px;

`
class Login extends React.Component<LoginProps, {}> {

    private userName: string = '';

    private handleLogin = () => {
        this.props.onLogin(this.userName);
    }

    private handleChange = (event: React.FormEvent<HTMLInputElement>) => {
        this.userName = event.currentTarget.value;
    }

    render() {
        const { userName } = this.props;
        return (
            <StyledLogin>
                {
                    userName ?
                        <WelcomeMessage>Hello {userName}</WelcomeMessage> :
                        <div>
                            <StyledLoginInput type='text' onChange={this.handleChange}></StyledLoginInput>
                            <LoginButton onClick={this.handleLogin}>login</LoginButton>
                        </div>
                }
            </StyledLogin>
        )
    }
}

export default Login;