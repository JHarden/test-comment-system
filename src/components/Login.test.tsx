import React from 'react';
import Login from './Login';
import renderer from 'react-test-renderer'
const mock = () => { }

const props = {
  userName: 'john',
  onLogin: () => {}
}

test('Login component test', () => {
  const component = renderer.create(
    <Login {...props}></Login>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
