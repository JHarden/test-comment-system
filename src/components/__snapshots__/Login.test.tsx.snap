// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Comment component test 1`] = `
<div
  style={
    Object {
      "color": "green",
      "outline": "1px solid green",
    }
  }
>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`Login component test 1`] = `
<div>
  <p>
    Login
  </p>
  <input
    onChange={[Function]}
    type="text"
  />
  <button
    onClick={[Function]}
  >
    click me
  </button>
</div>
`;
